message/validation/genesis/consensus_validation.go

```diff
@@ -131,8 +131,10 @@ func (mv *messageValidator) validateConsensusMessage(
 			signerState.Reset(msgRound)
 		}
 
-		if mv.hasFullData(signedMsg) && signerState.ProposalData == nil {
-			signerState.ProposalData = signedMsg.FullData
+		if signedMsg.Message.MsgType == genesisspecqbft.ProposalMsgType {
```

Comment:
what about round change?

Reply:
@GalRogozinski could you please confirm if it's correct?

Reply:
What you said is correct

If you look at https://github.com/ssvlabs/ssv-spec/blob/ccf408d1ecd87a4ab631885409d679f10c3fd080/qbft/round_change.go#L429 you should see that we add fullData to RC messages. The reason is because that if a node didn't see this data in the previous round, then on the next round it is the leader and is obliged to propose the same data, then it can get it from the RC message.

But this data must be justified in order to accept it as correct

Reply:
So if you have a quorum of valid RC messages that carry a justification of Prepares (quorum) for this data, then you know that the full data they carry must be the correct proposal data even if your node missed it

Reply:
@GalRogozinski thanks, so it means that this particular check should be done for both `genesisspecqbft.ProposalMsgType` and `genesisspecqbft.RoundChangeMsgType`?

Reply:
I see that on lines 210-212 you check for justifications if the signerState is nil...
so not sure
give me a call

Reply:
Per our conversation... the if check was added to avoid an attack where any node can send an RC message that changes the `mv` state.

The code is fine the way it is at the moment